src/error.rs

```diff
@@ -1153,6 +1162,26 @@ impl IntoDiagnostics<FileId> for EvalError {
                 arg_evaluated,
             )
             .into_diagnostics(files, stdlib_ids),
+            EvalError::QueryNonRecord { pos, id, value } => {
+                let label = format!(
+                    "tried to query field `{}`, but the expression has type {}",
+                    id,
+                    value
+                        .term
+                        .type_of()
+                        .unwrap_or_else(|| String::from("<unevaluated>")),
+                );
+
+                let labels = if let Some(span) = pos.into_opt() {
```

Comment:
(very) minor nitpick: if `vec![..]` is applied in both branches, you might as well factor it out, defining `let label = ...` as a single label and using `vec![label]` in `with_labels` below. It'll save a staggering number of 7 characters!